src/cr/cube/matrix/assembler.py

```diff
@@ -574,36 +574,43 @@ def table_margin(self):
-        # assembled, because an MR dimension can't have subtotals. Because the
-        # dimensionality of table-margin "follows" the MR-dimension(s), it follows that
-        # a table-margin vector or matrix never has subtotals.
-
-        base_table_margin = self._cube_result_matrix.table_margin
-
-        # --- MR_X_MR slice produces a 2D table-margin (each cell has its own N) ---
-        if (
-            self._rows_dimension.dimension_type == DT.MR_SUBVAR
-            and self._columns_dimension.dimension_type == DT.MR_SUBVAR
-        ):
-            return base_table_margin[np.ix_(self._row_order, self._column_order)]
+        if self._measures.table_margin.is_defined:
+            return self._measures.table_margin.value
+        if self._measures.columns_table_margin.is_defined:
+            return self._assemble_marginal(self._measures.columns_table_margin)
+        if self._measures.rows_table_margin.is_defined:
+            return self._assemble_marginal(self._measures.rows_table_margin)
+        return self.table_weighted_bases
 
-        # --- MR_X_CAT slice produces a 1D table-margin (each row has its own N) ---
-        if self._rows_dimension.dimension_type == DT.MR_SUBVAR:
-            return base_table_margin[self._row_order]
+    @lazyproperty
+    def table_margin_unpruned(self):
+        """np.float64 scalar or a 1D or 2D ndarray of np.float64 table margin.
 
-        # --- CAT_X_MR slice produces a 1D table-margin (each column has its own N) ---
-        if self._columns_dimension.dimension_type == DT.MR_SUBVAR:
-            return base_table_margin[self._column_order]
+        This value includes hidden vectors, those with either a hide transform on their
+        element or that have been pruned (because their base (N) is zero). Also, it does
+        not include inserted subtotals. This does not affect a scalar value but when the
+        return value is an ndarray, the shape may be different than the array returned
+        by `.table_margin`.
 
-        # --- CAT_X_CAT produces scalar table-margin ---
-        return base_table_margin
+        A matrix with a multiple-response (ARRAY) dimension produces an array of
+        table-margin values because each element (subvariable) of the dimension
+        represents a logically distinct question which may not have been asked of all
+        respondents. When both dimensions are ARRAY, the return value is a 2D ndarray. A
+        CAT_X_CAT matrix produces a scalar value for this property.
+        """
+        # TODO: see TODO in `.table_base_unpruned`
+        if self._measures.table_margin.is_defined:
+            return self._measures.table_margin.value
+        if self._measures.columns_table_margin.is_defined:
+            return self._measures.columns_table_margin.blocks[0]
+        if self._measures.rows_table_margin.is_defined:
+            return self._measures.rows_table_margin.blocks[0]
+        return self._measures.table_weighted_bases.blocks[0][0]
```

Comment:
I'm not seeing anything that take care of order ... does this property need to be underneath the sort-by-value logic?

Reply:
Looking closer, I think the `table_margin_unpruned` is intentionally unordered. If it were ordered than it would have to be pruned.